src/Client/AbstractClient.php

```diff
@@ -17,11 +19,14 @@ class AbstractClient
     private $baseUrl;
     /** @var string */
     private $apiPath = '/api/v1/';
+    /** @var CurlClient */
+    private $httpClient;
 
     public function __construct(string $baseUrl, string $apiKey)
     {
         $this->baseUrl = rtrim($baseUrl, '/');
         $this->apiKey = $apiKey;
+        $this->httpClient = new CurlClient();
```

Comment:
I guess you decided against allowing it to be set via the constructor. Fine with that I think although it is a bit strange to initialize it with the default CurlClient to then change it afterwards with the setter, but no hard feelings here.

Reply:
Yeah I couldn't really decide and because it would clutter up all constructions I ended up going this route.